src/jpeg/mod.rs

```diff
@@ -0,0 +1,20 @@
+//! Module for reading and recreation of JPEGs without the loss of any information.
+//!
+//! This means that it should be possible to reconstruct bit-by-bit an exactly identical
+//! JPEG file from the input.
+//!
+//! Note that we never actually decode the JPEG into pixels, since the DCT is lossy, so
+//! processing needs to be done at the DCT coefficient level and keep the coefficients in
+//! the BlockBasedImage identical.
```

Comment:
That is not completely true: effectively on DC prediction phase Lepton decodes each DCT block into pixels, so they can be extracted without encoding to JPEG back. Do you think it will be interesting to add such a possibility into the library?

Reply:
Yeah, so it just need to be added to all the block - and then we have pixels.

Reply:
the way the IDCT is used right now wouldn't be ideal of JPEG decoding... much faster if the decoding was done with 15 bit fixed point math that's more SIMD friendly. Also there's the subsampling and colorspace conversion that's a whole other thing.